agent/api/task.go

```diff
@@ -674,3 +678,27 @@ func (t *Task) String() string {
 	}
 	return res + "]"
 }
+
+// GetID is used to retrieve the taskID from taskARN
+// Reference: http://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html#arn-syntax-ecs
+func (task *Task) GetID() (string, error) {
```

Comment:
@petderek I've updated the error messages with more context. 
I want to use `errors.New` but I've combined it as `errors.New(fmt.Sprintf(...))`

Should we fallback to `fmt.Errorf` instead ?